toolchain/check/testdata/interop/cpp/no_prelude/file_not_found.carbon

```diff
@@ -0,0 +1,23 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/interop/cpp/no_prelude/file_not_found.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/interop/cpp/no_prelude/file_not_found.carbon
+// CHECK:STDERR: not_found.h: error: error opening file for read: No such file or directory [ErrorOpeningFile]
```

Comment:
This seems to be in the wrong place, it's not attached to the `import`?

Reply:
Yes, nice catch!
I think this is related to the bad diagnostics generated by `SourceBuffer`.
AFAIU, it puts the location of the diagnostic in the filename, which doesn't exist, so I assume this puts the diagnostic in the wrong place.
I plan to fix this by avoiding `SourceBuffer` when moving to importing C++ files by parsing a generated C++ code which includes files we want to import, though we'll probably need more iterations on making diagnostics better for C++ imports.
I hope this makes sense, please let me know otherwise and I'll work on fixing this.